### DISCUSSION TITLE
Callback from form setFieldValue

### DISCUSSION THREAD

**Discussion by MarkwinVI** [mantinedev/mantine#1513] [Q&A] [answered]
Hey, 

I'm trying to display in real-time how the user input changes based on the form values they enter, but whenever I set the values using setFieldValue and then try to get the form values they always are missing the latest input

i.e. 

I set the field value from "hey" to "hey1"
and when I  print out the form values I get that the value is still "hey"
when I add another input, only then I get back the previous value "hey1"

I assume that it's due to the form values not being updated yet, but is it possible to somehow await when all form values have been updated? 

(I tried setTimeout, but that didn't really work)

Thanks!

**Comment by rtivital** [comment, marked as answer]
`use-form` uses react useState hook to store and update values, unfortunately, this is how react works and there is no good way of avoiding that. In your case, I can recommend to either perform actions with the actual value in onChange handler:

```tsx
onChange={(value) => {
  form.setFieldValue('name', value);
  yourAction(value);
}}
```

or saving value to ref to use it anywhere else:

```tsx
onChange={(value) => {
  form.setFieldValue('name', value);
  valueRef.current = value;
}}

// ...later
yourAction(valueRef.current);
```

**Reply by MarkwinVI** [reply]
Thanks, I went a bit of a different route, not sure if its the recommended route, but it works

`function updateInputValue(key, value){
        form.setFieldValue(key, value);
        let formState = form.values;
        formState[key] = value;
        editComponet(formState)
    }`
